packages/db/index.js

```diff
@@ -890,63 +890,61 @@ export class DBClient {
 
     let query = this._client
       .from(psaPinRequestTableName)
-      .select(listPinsQuery)
+      .select(listPinsQuery, {
+        count: 'exact'
+      })
       .eq('auth_key_id', authKey)
       .is('deleted_at', null)
+      .limit(limit)
```

Comment:
Don't know if this is against the spec in any way, but if not we should consider adding a max to the limit (in the API). We do that for the [uploads as well[(https://github.com/web3-storage/web3.storage/blob/main/packages/api/src/user.js#L170), to avoid long queries that might overwhelm the system

Reply:
This is a really valid point and we're already tackling it as part of this [PR](https://github.com/web3-storage/web3.storage/pull/937/files#diff-9a666433f02a3fe427683b8f92fddccd7b1c9a9618d4b8b36b5f779b38559eaaR18) [here](https://github.com/web3-storage/web3.storage/pull/937/files#diff-9a666433f02a3fe427683b8f92fddccd7b1c9a9618d4b8b36b5f779b38559eaaR18)

Can we disregard it in the context of this one?

Reply:
Ok, let's track it in #937